docs/getting_started_with_jax_for_AI.md

```diff
@@ -0,0 +1,417 @@
+splits = train_test_split(digits.images, digits.target, random_state=0)
+```
+
+```{code-cell}
+:id: oMRcwKd4hqOo
+:outputId: 0ad36290-397b-431d-eba2-ef114daf5ea6
+
+import jax.numpy as jnp
+images_train, images_test, label_train, label_test = map(jnp.asarray, splits)
+print(f"{images_train.shape=} {label_train.shape=}")
+print(f"{images_test.shape=}  {label_test.shape=}")
+```
+
++++ {"id": "JzrixENjifiq"}
+
+### Defining the flax model
+
+We can now use the [Flax](http://flax.readthedocs.io) package to create a simple [Feedforward](https://en.wikipedia.org/wiki/Feedforward_neural_network) neural network withone hidden layer, and use a *scaled exponential linear unit* (SELU) activation function:
+
+```{code-cell}
+:id: U77VMQwRjTfH
+
+from flax import nnx
+
+N_FEATURES = 64  # Number of pixels.
+N_INTERMEDIATE = 100  # Hidden layer size
+N_TARGETS = 10  # Number of categories.
+
+class SimpleNN(nnx.Module):
+
+  def __init__(self, *, rngs: nnx.Rngs):
+    self.layer1 = nnx.Linear(N_FEATURES, N_INTERMEDIATE, rngs=rngs)
+    self.layer2 = nnx.Linear(N_INTERMEDIATE, N_INTERMEDIATE, rngs=rngs)
+    self.layer3 = nnx.Linear(N_INTERMEDIATE, N_TARGETS, rngs=rngs)
+
+  def __call__(self, x):
+    x = x.reshape(-1, 64)  # flatten 8x8 images
+    x = nnx.selu(self.layer1(x))
+    x = nnx.selu(self.layer2(x))
+    x = self.layer3(x)
+    return x
+
+model = SimpleNN(rngs=nnx.Rngs(0))
+```
+
++++ {"id": "FIXmNs5-lrEf"}
+
+### Training the model
+
+With this model defined, we can now use the [optax](http://optax.readthedocs.io) package to train it.
+First we define a loss function using [`optax.softmax_cross_entropy_with_integer_labels`](https://optax.readthedocs.io/en/latest/api/losses.html#optax.losses.softmax_cross_entropy_with_integer_labels), and then a training step based on this optimizer:
+
+```{code-cell}
+:id: QwRvFPkYl5b2
+
+import jax
+import optax
+
+optimizer = nnx.Optimizer(model, optax.sgd(learning_rate=0.05))
+
+def loss_fun(
+    model: nnx.Module,
+    data: jax.Array,
+    labels: jax.Array):
+  logits = model(data)
+  loss = optax.softmax_cross_entropy_with_integer_labels(
+    logits=logits, labels=labels
+  ).mean()
+  return loss, logits
+
+@nnx.jit  # JIT-compile the function
+def train_step(
+    model: nnx.Module,
+    optimizer: nnx.Optimizer,
+    data: jax.Array,
+    labels: jax.Array):
+  loss_gradient = nnx.value_and_grad(loss_fun, has_aux=True)  # gradient transform!
+  (loss, logits), grads = loss_gradient(model, data, labels)
+  optimizer.update(grads)  # inplace update
+```
+
++++ {"id": "K2Tp-ym6sXEl"}
+
+Notice here the use of `nnx.jit` and `nnx.grad`, which are transformations built on `jax.jit` and `jax.grad`: `jit` is a Just In Time compilation transformation, and will cause the function to be passed to the XLA compiler for fast repeated execution. `grad` is a gradient transformation, and uses JAX's automatic differentiation for fast optimization of large networks.
```

Comment:
One thing we discussed is the right place to talk about debugging and best practices for how to do debugging of jitted functions.  Since we just started using JIT here we should discuss some simple options briefly and then link to where it is discussed in more depth.